File: src/main/java/io/github/takenoha/towerdefense/persistence/TowerRepairMutationResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;

/** Atomic result of a battle-funds tower repair. */
public record TowerRepairMutationResult(
        OperationOutcome outcome,
        TowerDurability durability,
        BattleFunds funds) {
    public TowerRepairMutationResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(durability, "durability");
        Objects.requireNonNull(funds, "funds");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/TowerRepairMutationResult.kt
package io.github.takenoha.towerdefense.persistence

import java.util.Objects
import kotlin.jvm.JvmRecord

/** Atomic result of a battle-funds tower repair. */
@JvmRecord
data class TowerRepairMutationResult(
    val outcome: OperationOutcome,
    val durability: TowerDurability,
    val funds: BattleFunds,
) {
    init {
        Objects.requireNonNull(outcome, "outcome")
        Objects.requireNonNull(durability, "durability")
        Objects.requireNonNull(funds, "funds")
    }
}
